### PR TITLE
Allow config dir path to be passed via command line

### LINKED ISSUE #876: Create commandline interface for comit-rs and btsieve
Create a CLI for the comit-node and btsieve to customize the location of the configuration file. 

The go-to library for creating CLIs in Rust is [structopt](https://crates.io/crates/structopt).

DoD: 
- [x] cli for comit_node
- [x] cli for btsieve

### PR DESCRIPTION
**Pull request by D4nte**
If no argument passed, it falls back to ~/.config/comit_node

Once this is approved/merged, I'll also do it for btsieve

Resolves #876

### CODE REVIEW

**Comment by D4nte** [comment]
I am slightly confused. @comit-network/rust-devs do we want to remove the env variable all together?

**Comment by thomaseizinger** [comment]
> @comit-network/rust-devs do we want to remove the env variable all together?

Yes. As far as I remember, we concluded that env variables are cumbersome to work with + it is a bad choice for overriding the config file path.

IMO, we may want to use env variables at some point but not for the config file path.

**Comment by bonomat** [comment]
> Yes. As far as I remember, we concluded that env variables are cumbersome to work with + it is a bad choice for overriding the config file path.

Did we conclude that? :D I know the ticket says so, but I think to remember we want to look for a fallback environment variable if the commandline argument was not passed in.

File: README.md [hunk 1]
```diff
@@ -37,7 +37,10 @@ Contains crates specific to our application. Can depend on libraries located in
 ## Build & Run
 
 1. `cargo build` (do `export SOLC_BIN=/usr/bin/solc` if `solc` is installed locally)
-2. Put a [`default.toml`](application/comit_node/config/default.toml) config file into `~/.config/comit_node` or set `COMIT_NODE_CONFIG_PATH` as folder path to where the `default.toml` is located.
+2. Put a [`default.toml`](application/comit_node/config/default.toml) config file into `~/.config/comit_node`
+   - or pass `--config <config_path>`
+   - or set `COMIT_NODE_CONFIG_PATH=<config_path>`
```
[the remark this answers is not in the document]
Comment (D4nte) [reply README.md]: the logo to use is `:mega:` :)

File: application/comit_node/src/lib.rs [hunk 2]
```diff
@@ -8,9 +8,27 @@ pub mod comit_client;
...
 pub mod network;
 pub mod node_id;
 pub mod seed;
 pub mod settings;
 pub mod swap_protocols;
+
+fn var_or_default(name: &str, default: String) -> String {
+    match std::env::var(name) {
+        Ok(value) => {
+            log::info!("Set {}={}", name, value);
```
Comment (bonomat) [inline application/comit_node/src/lib.rs]: Shall we set a different log level for this, e.g. `debug` or `trace`?

Comment (D4nte) [reply application/comit_node/src/lib.rs]: I'd keep it like this for now as it says where comit_node get the config from.
It'll then get trashed with #947.

Comment (bonomat) [reply application/comit_node/src/lib.rs]: True. I keep on forgetting about the other ticket...

File: application/comit_node/src/lib.rs [hunk 3]
```diff
@@ -8,9 +8,27 @@ pub mod comit_client;
...
 pub mod libp2p_bam;
+pub mod load_settings;
 pub mod logging;
 pub mod network;
 pub mod node_id;
 pub mod seed;
 pub mod settings;
 pub mod swap_protocols;
+
+fn var_or_default(name: &str, default: String) -> String {
```
Comment (thomaseizinger) [inline application/comit_node/src/lib.rs]: Hopefully this thing can just go away soon!

**Pull request merged**